azurerm/internal/services/compute/resource_arm_disk_encryption_set.go

```diff
@@ -0,0 +1,387 @@
+			resp, err := client.Get(ctx, resourceGroup, keyVaultName)
+			if err != nil {
+				return fmt.Errorf("Error reading keyvault %q (Resource Group %q): %+v", keyVaultName, resourceGroup, err)
+			}
+			if props := resp.Properties; props != nil {
+				if softDelete := props.EnableSoftDelete; softDelete != nil {
+					if !*softDelete {
+						return fmt.Errorf("the keyvault in Disk Encryption Set must enable soft delete")
+					}
+				}
+				if purgeProtection := props.EnablePurgeProtection; purgeProtection != nil {
+					if !*purgeProtection {
+						return fmt.Errorf("the keyvault in Disk Encryption Set must enable purge protection")
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func resourceArmDiskEncryptionSetRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.DiskEncryptionSetsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["diskEncryptionSets"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Disk Encryption Set %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Disk Encryption Set %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if encryptionSetProperties := resp.EncryptionSetProperties; encryptionSetProperties != nil {
+		if err := d.Set("active_key", flattenArmDiskEncryptionSetKeyVaultAndKeyReference(encryptionSetProperties.ActiveKey)); err != nil {
+			return fmt.Errorf("Error setting `active_key`: %+v", err)
+		}
+		if err := d.Set("previous_keys", flattenArmDiskEncryptionSetKeyVaultAndKeyReferenceArray(encryptionSetProperties.PreviousKeys)); err != nil {
+			return fmt.Errorf("Error setting `previous_keys`: %+v", err)
+		}
+	}
+	if identity := resp.Identity; identity != nil {
+		if err := d.Set("identity", flattenArmDiskEncryptionSetIdentity(identity)); err != nil {
+			return fmt.Errorf("Error setting `identity`: %+v", err)
+		}
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmDiskEncryptionSetDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.DiskEncryptionSetsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["diskEncryptionSets"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
```

Comment:
we should be able to remove this, since this'll get caught by the Read

Reply:
Done